### ISSUE TITLE
--warp flag issue

### ISSUE THREAD

**Issue by PhilosopherRex** [paritytech/parity#6895]
> I'm running:
>
> - **Parity version**: 1.7.7
> - Linux - debian
> - binaries

This is probably (almost certainly) not an issue ... I've most likely caused the problem, but don't know how to resolve.

I'm unable to upgrade because 1.8.1 (and maybe 1.8.0) don't like the --warp flag. 

I get "error: Found argument '--warp' which wasn't expected, or isn't valid in this context"

Somehow (I believe) I've set a config *somewhere* that has that flag in it ... the problem is I've checked all the locations for parity config files on Linux and there is no '--warp' or any other flags listed  ... so I'm guessing there must be some obscure *default* config file and I managed to find it (months ago) and adjusted it stupidly and now it defaults to flagging '--warp' whenever I enter the command parity.

So my question is, where is the default config file located or even a name for it?  I've looked everywhere my tiny brain can think of.

cheers

---

**Comment by Office-Julia** [comment]
Hello. Probably this answers your question:

> The --warp flag literally does nothing because warp is enabled by default. So it does not affect your connected peers in any way. It must be a coincidence. You can try to increase your minimum number of peers with --min-peers and add additional snapshot clients with --snapshot-peers.
> 
> So, to answer your initial question how geth --warp relates to geth --fast, yes, geth' fast sync mode does only work on first run, parity's warp sync works always if it detects the chain is a catch up a lot of blocks.
> 
> 

Reference: https://ethereum.stackexchange.com/questions/16750/how-is-paritys-warp-sync-related-to-geths-fast-sync

Let me know if you have any further issues.

**Comment by 5chdn** [comment]
Probably related to #6906

Does this prevent you from running parity? Do you use a configuration file?

**Comment by PhilosopherRex** [comment]
Yes, but I can run the older version of parity, but the newer version fails with the error message as I stated above.  I have discovered I can run it as another user though, so it must be a config file somewhere, but it's not the normal one ~/.local/share/io.parity.ethereum/config.toml   -- which is empty in my case.

**Comment by axelchalon** [comment]
The `--warp` flag was removed with this PR https://github.com/paritytech/parity/pull/6356. I think that before this the flag was just ignored with a warning.

Incidentally, I just found out that the Windows installer automatically sets `--warp` as a launch option for Parity: https://github.com/paritytech/parity/blob/master/nsis/installer.nsi#L14 I'll make a PR to fix this; but I don't think this is related to your issue as you're running Linux.
